### PR TITLE
changed formatting changes

### PR DESCRIPTION
**Pull request by lisham2000**

### CODE REVIEW

File: nion/swift/ExportDialog.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+import enum
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:3]: This should be before `import gettext` below. The imports are all in the same section and alphabetized.

File: nion/swift/ExportDialog.py [hunk 2]
```diff
@@ -33,7 +34,8 @@
     from nion.swift.model import DisplayItem
 
 _ = gettext.gettext
-
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:36]: Don't remove this space. The constants should be declared after a blank line.

File: nion/swift/ExportDialog.py [hunk 3]
```diff
@@ -33,7 +34,8 @@
...
+CM_PER_INCH = 2.54
+PIXELS_PER_INCH = 96
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:39]: There should be exactly two blank lines before each top-level class. Please add one here.

File: nion/swift/ExportDialog.py [hunk 4]
```diff
@@ -221,56 +223,163 @@ def cancel(self) -> bool:
...
-    def __init__(self, display_size: Geometry.IntSize) -> None:
-        self.width_model = Model.PropertyModel(display_size.width)
-        self.height_model = Model.PropertyModel(display_size.height)
+class UnitType(enum.Enum):
+    INCHES = 0
+    CENTIMETERS = 1
+    PIXELS = 2
 
-        self.int_converter = Converter.IntegerToStringConverter()
+class ExportSVGHandler:
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:231]: There should be exactly two blank lines before each top-level class. Please add one here.

File: nion/swift/ExportDialog.py [hunk 5]
```diff
@@ -221,56 +223,163 @@ def cancel(self) -> bool:
...
+            return Geometry.FloatSize(height=height_px / PIXELS_PER_INCH, width=width_px / PIXELS_PER_INCH)
+        elif unit_type == UnitType.CENTIMETERS:
+            return Geometry.FloatSize(height=height_px * CM_PER_INCH / PIXELS_PER_INCH,
+                                      width=width_px * CM_PER_INCH / PIXELS_PER_INCH)
+        return Geometry.FloatSize(width=width_px, height=height_px)  # Pixels
+
     def close(self) -> None:
         pass
 
-
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:241]: There should be exactly two blank lines before each top-level class. Please add this one back.

File: nion/swift/ExportDialog.py [hunk 6]
```diff
@@ -221,56 +223,163 @@ def cancel(self) -> bool:
...
 class ExportSVGDialog:
-
-    def __init__(self, document_controller: DocumentController.DocumentController, display_item: DisplayItem.DisplayItem) -> None:
+    def __init__(self, document_controller: DocumentController.DocumentController,
+                 display_item: DisplayItem.DisplayItem) -> None:
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:351]: Please revert this unnecessary formatting change.

File: nion/swift/ExportDialog.py [hunk 7]
```diff
@@ -221,56 +223,163 @@ def cancel(self) -> bool:
...
+            path, selected_filter, selected_directory = document_controller.get_save_file_path(_("Export File"),
+                                                                                               export_dir, filter, None)
             if path and not os.path.splitext(path)[1]:
                 path = path + os.path.extsep + "svg"
             if path:
                 ui.set_persistent_string("export_directory", selected_directory)
-                display_shape = Geometry.IntSize(height=height_px, width=width_px)
-                document_controller.export_svg_file(DisplayPanel.DisplayPanelUISettings(ui), display_item, display_shape, pathlib.Path(path))
+                document_controller.export_svg_file(DisplayPanel.DisplayPanelUISettings(ui), display_item,
+                                                    display_shape, pathlib.Path(path))
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:382]: Please revert this unnecessary formatting change.

File: nion/swift/ExportDialog.py [hunk 8]
```diff
@@ -221,56 +223,163 @@ def cancel(self) -> bool:
...
+                ),
+                handler.previous_unit_index
+            )
             ui = document_controller.ui
             filter = "SVG File (*.svg);;All Files (*.*)"
             export_dir = ui.get_persistent_string("export_directory", ui.get_document_location())
             export_dir = os.path.join(export_dir, display_item.displayed_title)
-            path, selected_filter, selected_directory = document_controller.get_save_file_path(_("Export File"), export_dir, filter, None)
+            path, selected_filter, selected_directory = document_controller.get_save_file_path(_("Export File"),
+                                                                                               export_dir, filter, None)
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:376]: Please revert this unnecessary formatting change.

File: nion/swift/ExportDialog.py [hunk 9]
```diff
@@ -221,56 +223,163 @@ def cancel(self) -> bool:
...
-    def __init__(self, document_controller: DocumentController.DocumentController, display_item: DisplayItem.DisplayItem) -> None:
+    def __init__(self, document_controller: DocumentController.DocumentController,
+                 display_item: DisplayItem.DisplayItem) -> None:
         super().__init__()
-
         self.__document_controller = document_controller
-
         u = Declarative.DeclarativeUI()
-
+        u = Declarative.DeclarativeUI()
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:355]: We only need one `u = ...` statement here!

File: nion/swift/ExportDialog.py [hunk 10]
```diff
@@ -221,56 +223,163 @@ def cancel(self) -> bool:
...
-
+            display_size = Geometry.IntSize(height=3, width=5)
         handler = ExportSVGHandler(display_size)
 
         def ok_clicked() -> bool:
-            dpi = 96
-            width_px = (handler.width_model.value or display_size.width) * dpi
-            height_px = (handler.height_model.value or display_size.height) * dpi
-
+            display_shape: Geometry.IntSize = handler.convert_to_pixels(
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:364]: The `display_shape: Geometry.IntSize. = ` should just be `display_shape = `. The type checker should be able to infer the type from the return value of `handler.convert_to_pixels`.

File: nion/swift/ExportDialog.py [hunk 11]
```diff
@@ -221,56 +223,163 @@ def cancel(self) -> bool:
...
+            pass  # Handle the case where the input is not a valid integer
+
+    def unit_changed(self, widget: UserInterface.ComboBoxWidget, current_index: int) -> None:
+        new_unit_type = UnitType(current_index)
+        self.size = self.convert_to_pixels(
+            Geometry.FloatSize(height=float(self.height_value_line_edit.text or 0),
+                               width=float(self.width_value_line_edit.text or 0)),
+            self.previous_unit_index
+        )
+        new_width, new_height = self.convert_from_pixels(self.size, new_unit_type)
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:309]: This line is wrong:
`new_width, new_height = self.convert_from_pixels(self.size, new_unit_type)`

It should be `new_shape = self.convert_from_pixels(self.size, new_unit_type)` and then use `new_shape.width` and `new_shape.height`.

In the incorrect code, you assign the `FloatSize` returned by `self.convert_from_pixels` to two variables and a Python language feature called 'unpacking' occurs where it assigns the return value subscripted by each unpacked value into the variables. So if the result were in `r`, `new_width` would get `r[0]` and `new_height` would get `r[1]`. This is wrong since index `0` is for height. A quick test of the dialog with the line plot reveals this bug.

File: nion/swift/ExportDialog.py [hunk 12]
```diff
@@ -221,56 +223,163 @@ def cancel(self) -> bool:
...
+        self.previous_unit_index = UnitType.PIXELS
+        self.width_value_line_edit: UserInterface.LineEditWidget
+        self.height_value_line_edit: UserInterface.LineEditWidget
+        self.size = Geometry.IntSize(0, 0)  # Changed from FloatSize to IntSize
         u = Declarative.DeclarativeUI()
-        width_row = u.create_row(u.create_label(text=_("Width (in)"), width=80), u.create_line_edit(text="@binding(width_model.value, converter=int_converter)"), spacing=12)
-        height_row = u.create_row(u.create_label(text=_("Height (in)"), width=80), u.create_line_edit(text="@binding(height_model.value, converter=int_converter)"), spacing=12)
-        main_page = u.create_column(width_row, height_row, spacing=12, margin=12)
-
+        self.width_line_edit = u.create_line_edit(
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:245]: `self.width_line_edit`, `self.height_line_edit`, and `self.units_combo_box` are only used in this function and so should be local variables.

However. I'd really like to not use intermediate variables when not necessary as it makes the structure of the dialog more difficult to understand. Please consider doing something like here:

https://github.com/nion-software/nionswift/blob/b21329ff63f00027ba74e43fe0ee693b6de79e74/nion/swift/Inspector.py#L2347-L2368

**Review by cmeyer — CHANGES_REQUESTED** [review]
Thanks for the (mostly fixed) formatting changes. Unfortunately there are several unnecessary formatting changes still there. I tried to mark the remaining ones. I also made other comments to address, including a bug about switching width/height.

Also, a useful commit message would be better. see [pull request guidelines](https://github.com/nion-software/nionswift/wiki/Contributing#pull-request-guidelines) and [writing good commit messages](https://github.blog/2022-06-30-write-better-commits-build-better-projects/). The current message is "changed formatting changes" and contains no useful information. Instead, it could be something like "Expanded SVG export to allow size to be specified in inches or centimeters".

Once these changes are made, I'll continue the review with some more structural changes.

File: nion/swift/ExportDialog.py (hunk 1 above)
Comment (cmeyer) [reply nion/swift/ExportDialog.py]: We should also remove the unused `import functools`. I had deleted that in my local copy, but if it were to remain, `import enum` should go (alphabetically) before `import functools`.

File: nion/swift/ExportDialog.py (hunk 6 above)
Comment (cmeyer) [reply nion/swift/ExportDialog.py:348]: This is still an unnecessary formatting change.

File: nion/swift/ExportDialog.py (hunk 7 above)
Comment (cmeyer) [reply nion/swift/ExportDialog.py:377]: Same here.

File: nion/swift/ExportDialog.py (hunk 8 above)
Comment (cmeyer) [reply nion/swift/ExportDialog.py]: This was almost fixed, but now there is a removed space. Needs another pass.

File: nion/swift/ExportDialog.py [hunk 13]
```diff
@@ -221,56 +226,155 @@ def cancel(self) -> bool:
...
+                    width=float(handler.width_model.value or 0)
+                ),
+                handler.previous_unit_index
+            )
             ui = document_controller.ui
             filter = "SVG File (*.svg);;All Files (*.*)"
             export_dir = ui.get_persistent_string("export_directory", ui.get_document_location())
             export_dir = os.path.join(export_dir, display_item.displayed_title)
-            path, selected_filter, selected_directory = document_controller.get_save_file_path(_("Export File"), export_dir, filter, None)
+            path, selected_filter, selected_directory = document_controller.get_save_file_path(_("Export File"),export_dir, filter, None)
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:371]: Still an unnecessary formatting change, now missing a space that was there before.

File: nion/swift/ExportDialog.py [hunk 14]
```diff
@@ -221,56 +226,155 @@ def cancel(self) -> bool:
...
+        self.size = self.convert_to_pixels(
+            Geometry.FloatSize(height=float(height_text or 0),
+                               width=float(width_text or 0)),
+            self.previous_unit_index
+        )
+        newshape = self.convert_from_pixels(self.size, new_unit_type)
+
+        if new_unit_type == UnitType.PIXELS:
+            new_width:float = round(newshape.width)
+            new_height:float = round(newshape.height)
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:314]: These ":float" parts should be unnecessary and should be removed.

Also, if they were required for some reason, please a space after the ":":
`new_width: float = ...`

File: nion/swift/ExportDialog.py [hunk 15]
```diff
@@ -1,7 +1,9 @@
 from __future__ import annotations
 
+
 # standard libraries
 import functools
+import enum
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:6]: Sorry it looks like it's still not alphabetical (my instructions were wrong). But `import functools` can be removed (for now) since it isn't used - and doing that will finally make this group of imports alphabetical.

**Review by cmeyer — CHANGES_REQUESTED** [review]
Better, but still several formatting issues.

**Comment by lisham2000** [comment]
Closed due to revised version at #1106
